prody/dynamics/plotting.py

```diff
@@ -2381,3 +2355,43 @@ def showTree_networkx(tree, node_size=20, node_color='red', node_shape='o',
         showFigure()
 
     return mpl.gca()
+
+
+def checkColors(colors, num, labels, allowNumbers=False):
+    """Check colors and process them if needed"""
+
+    from matplotlib.colors import is_color_like
+
+    colors_dict = {}
+
+    if isinstance(colors, np.ndarray):
+        colors = tuple(colors)
+
+    if is_color_like(colors) or colors is None:
+        colors = [colors] * num
+    elif isListLike(colors):
+        colors = list(colors)
+
+    if isinstance(colors, dict):
+        if labels is None:
+            raise TypeError('color must be a string or a list unless labels are provided')
+        colors_dict = colors
+        colors = [colors_dict[label] for label in labels]
+
+    if isinstance(colors, list):
+        if len(colors) != num and not is_color_like(colors):
```

Comment:
I don't think we need `and not is_color_like(colors)` here since if it was color-like, line 2371 would have converted it to a list of colors (of `num`) already (correct me if I am wrong).